### PR TITLE
feat: use ko in kuttl e2e

### PR DESCRIPTION
**Pull request by weisdd**
- Moves from docker builds to ko for e2e tests.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -18,36 +16,39 @@ jobs:
...
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2.4.1
+          version: v0.13.0
+
       - name: Set up KinD
         id: kind
         run: |
           kind create cluster --image=kindest/node:v1.25.3 --config tests/e2e/kind.yaml
+
+      - name: Set up Docker Buildx
```
Comment (NissesSenap) [inline .github/workflows/e2e.yaml:34]: Why is docker buildx still needed?

**Comment by NissesSenap** [comment]
Should we remove the container image as well? Or we can do that in a separate PR where we also update the contribution guide.

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (weisdd) [reply .github/workflows/e2e.yaml:34]: When I tried to make `ko` load image directly to kind on mac, it failed, so just made it load image to docker daemon and then explicitly pass it to kind `kind load docker-image`. Direct load might work on runner / Linux, IDK.
We don't need buildx per say, pure docker is enough as well.

**Comment by weisdd** [comment]
@NissesSenap Yeah, I think that it's better to leave `Dockerfile` until we have an up-to-date contribution guide.

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (NissesSenap) [reply .github/workflows/e2e.yaml:34]: Okay, well It looks like it works. Really nice, I will merge it

**Pull request merged**